src/main/java/ttt/Board.java

```diff
@@ -2,149 +2,108 @@
-            grid[cell] = emptyCell;
+            board[cell] = emptyCell;
         }
     }
 
-    public void placeMark(String symbol, int location) {
-        if (!isValidLocation(location)) {
-            this.status = "invalid location";
-        } else if (isCellTaken(location)) {
-            this.status = "taken";
-        } else {
-            grid[location] = symbol;
-            status = "mark placed";
-        }
-    }
-
-    public String getStatus() {
-        return status;
+    public void placeMark(Marks mark, int location) {
+        board[location] = mark;
     }
 
     public int size() {
-        return grid.length;
-    }
-
-    public String get(int cell) {
-        return grid[cell];
-    }
-
-    public String getWinningMark() {
-        return winningMark;
+        return board.length;
     }
 
-    public boolean isAWinFor(String mark) {
-        return isWon() && winningMark.equals(mark);
+    public Marks getMarkAt(int cell) {
+        return board[cell];
     }
 
-    public void clear (int location) {
-        grid[location] = emptyCell;
+    public void clear(int location) {
+        board[location] = emptyCell;
     }
 
     public List<Integer> availableMoves() {
         List<Integer> moves = new ArrayList<>();
-        for (int cell = 0; cell < grid.length; cell ++) {
-            if (grid[cell].equals(emptyCell)) {
-               moves.add(cell);
-            }
+        for (int cell = 0; cell < board.length; cell++) {
+            if (board[cell].equals(emptyCell)) moves.add(cell);
         }
         return moves;
     }
 
-    public List<List<String>> winningPositions() {
-        List<List<String>> positions = new ArrayList<>();
-        positions.addAll(rows());
-        positions.addAll(columns());
-        positions.addAll(diagonals());
-        return positions;
-    }
-
-
     public boolean isFull() {
-        for (String cell : grid) {
-            if (cell.equals(emptyCell)) {
-                return false;
-            }
-        }
+        for (Marks cell : board) if (cell.equals(emptyCell)) { return false; }
         return true;
     }
 
     public boolean isEmpty() {
-        for (String cell : grid) {
-            if (!cell.equals(emptyCell)) {
-                return false;
-            }
-        }
+        for (Marks cell : board) if (!cell.equals(emptyCell)) { return false; }
         return true;
     }
 
     public boolean isDraw() {
-        return (isFull() && !isWon());
+        return isFull() && !isWon();
     }
 
     public boolean isFinished() {
-        return (isDraw() || isWon());
+        return isDraw() || isWon();
     }
 
-    public boolean isWon() {
-        for (List<String> selection : winningPositions()) {
-            if (isAWin(selection)) {
-                return true;
-            }
+    public boolean isAWinFor(Marks mark) {
+        for (List<Marks> cells : winningPositions()) {
+            if (Collections.frequency(cells, mark) == 3) return true;
```

Comment:
I would extract `Collection.frequency(cells, mark) == 3` into its own little method, as you use it a couple of more times down the file...